samples/sensor/ltc4162/src/main.c

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (c) 2021 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/sensor.h>
+#include <sys/printk.h>
+
+static const char *get_charger_status(enum charger_status value)
+{
+	switch (value) {
+	case CHARGER_STATUS_CHARGING:
+		return "charging";
+	case CHARGER_STATUS_NOT_CHARGING:
+		return "not charging";
+	case CHARGER_STATUS_UNKNOWN:
+		return "unknown";
+	default:
+		return NULL;
+	}
+}
+
+static const char *get_charger_type(enum charge_type type)
+{
+	switch (type) {
+	case CHARGER_CHARGE_TYPE_NONE:
+		return "not known";
+	case CHARGER_CHARGE_TYPE_FAST:
+		return "fast";
+	case CHARGER_CHARGE_TYPE_TRICKLE:
+		return "trickle";
+	default:
+		return NULL;
+	}
+}
+
+static const char *get_charger_health(enum charger_health health)
+{
+	switch (health) {
+	case CHARGER_HEALTH_UNSPEC_FAILURE:
+		return "unspec failure";
+	case CHARGER_HEALTH_DEAD:
+		return "dead";
+	case CHARGER_HEALTH_GOOD:
+		return "good";
+	default:
+		return NULL;
+	}
+}
+
+static void ltc4162_show_values(const char *type, struct sensor_value value)
+{
+	if ((value.val2 < 0) && (value.val1 >= 0)) {
+		value.val2 = -(value.val2);
+		printk("%s: -%d.%06d\n", type, value.val1, value.val2);
+	} else if ((value.val2 > 0) && (value.val1 < 0)) {
+		printk("%s: %d.%06d\n", type, value.val1, value.val2);
+	} else if ((value.val2 < 0) && (value.val1 < 0)) {
+		value.val2 = -(value.val2);
+		printk("%s: %d.%06d\n", type, value.val1, value.val2);
+	} else {
+		printk("%s: %d.%06d\n", type, value.val1, value.val2);
+	}
+}
+
+static void process_sample(const struct device *dev)
+{
+	const char *charger_status, *charger_type, *charger_health;
+	struct sensor_value chrg_status, chrg_type,
+	       input_current_limit, charger_temperature,
+	       const_chrg_current, const_chrg_current_max,
+	       chrg_health, input_voltage, input_current,
+	       const_chrg_voltage, const_chrg_voltage_max;
+
+	if (sensor_sample_fetch_chan(dev, SENSOR_CHAN_CHARGER_STATUS) < 0) {
+		printk("Failed to fetch charger status\n");
+		return;
+	}
+
+	if (sensor_channel_get(dev, SENSOR_CHAN_CHARGER_STATUS,
+			       &chrg_status) < 0) {
+		printk("Can't read LTC4162 Charger Status channel\n");
+		return;
+	}
+
+	charger_status = get_charger_status(chrg_status.val1);
+	printk("Charger_status: %s\n", charger_status);
+
+	if (sensor_sample_fetch_chan(dev, SENSOR_CHAN_CHARGER_TYPE) < 0) {
+		printk("Failed to fetch charger type\n");
```

Comment:
suggestion: if `printk` prints the error code returned by `sensor_...` calls, problems could be easier to debug.

```c
int ret;

ret = sensor_sample_fetch_chan(dev, SENSOR_CHAN_CHARGER_TYPE);
if (ret < 0) {
		printk("Failed to fetch charger type (%d)\n", ret);
		return;
}
```